backend/src/packages/chaiNNer_pytorch/__init__.py

```diff
@@ -115,8 +116,68 @@ def get_pytorch():
             size_estimate=36.5 * KB,
         ),
     ],
+    icon="PyTorch",
+    color="#DD6B20",
+)
+
+package.add_setting(
+    ToggleSetting(
+        label="CPU Mode",
+        key="cpu_mode",
+        description="Use CPU for PyTorch instead of GPU. This is much slower and not recommended.",
+        default=False,
+    ),
+)
+
+package.add_setting(
+    ToggleSetting(
+        label="FP16 Mode",
+        key="fp16_mode",
+        description=(
+            "Runs PyTorch in half-precision (FP16) mode for less RAM usage."
+            if is_arm_mac
+            else "Runs PyTorch in half-precision (FP16) mode for less VRAM usage. RTX GPUs also get a speedup."
+        ),
+        default=False,
+    ),
+)
+
+if not is_arm_mac:
+    gpu_list = []
+    try:
+        import torch
+
+        for i in range(torch.cuda.device_count()):
+            device_name = torch.cuda.get_device_properties(i).name
+            gpu_list.append(device_name)
+    except:
+        pass
+
+    package.add_setting(
+        DropdownSetting(
+            label="GPU",
+            key="gpu",
+            description="Which GPU to use for PyTorch. This is only relevant if you have multiple GPUs.",
+            options=[{"label": x, "value": str(i)} for i, x in enumerate(gpu_list)],
+            default="0",
+            disabled=len(gpu_list) <= 1,
+        )
+    )
+
+PyTorchSettings = TypedDict(
+    "pytorch_settings",
+    {
+        "cpu_mode": bool,
+        "fp16_mode": bool,
+        "gpu": int,
+    },
 )
 
+
+def get_pytorch_settings() -> PyTorchSettings:
+    return cast(PyTorchSettings, package.get_execution_settings())
```

Comment:
I really don't like this. It's both wrong (`gpu` isn't defined for ARM macs) and inconvenient to use (`.get("some-key", default)`; I hope you use the same default everywhere). 

Instead, I think that `get_settings` should parse the raw settings dict. Combined with my `settings.py` suggestions, we could have something like this:

```py
# settings.py
import torch
from . import package

package.add_setting(...)

@dataclass
class PyTorchSettings:
    cpu_mode: bool
    fp16: bool
    gpu: int

    def get_device(self) -> torch.device: ...

def get_settings() -> PyTorchSettings:
    raw = package.get_execution_settings()
    return PyTorchSettings(
        cpu_mode=raw.get("cpu_mode", false),
        ...
    )
```

This also adds a nice abstraction layer between the way we internally store settings and the way nodes consume settings.